Project completion

edgardsoza opened the pull request:
This is my capstone project for module 1 in Microverse. 
1. Home and About pages, created.
2. Mobile and desktop versions created. 
3. Home and about pages links working.
4. Mobile menu when users taps on hamburger, working.
5. Close button is working as well. 
6. Feature speakers was created with JS. 
7. Link for the loom video here: https://www.loom.com/share/374cd11aa5e94c42bc8fd69f2d559e30.
8. README file created and edited accordingly.

AbayomiOlaoye commented on about.html:
```diff
@@ -0,0 +1,117 @@
+            <img src="images/airbnb-partner.jpg" alt="Airbnb logo" class="partner-image">
+        </div>
+    </section>
+    <footer class="footer-about">
+        <div class="footer-container">
+        <div class="footer-about-image-container">
+        <img src="images/genmexlogo.jpg" alt="Footer logo About" class="footer-logo-about">
+        </div>
+        <div>
+            <p class="license-about">Gen Mex Concert 2023 Amazon Originals<br>Some Rights Reserved</p>
+        </div>
+        </div>
+    </footer>
```
- [x] I wish I was able to get a screenshot of the strange white space after this footer section in the desktop view but strangely, it's not working. Kindly check to improve your design by removing this with space :+1:

edgardsoza replied:
Solved.

AbayomiOlaoye commented on index.html:
```diff
@@ -0,0 +1,144 @@
+    <div class="nav-bar display-none">
+        <div class="nav-bar-container">
+        <a href="./index.html" class="logo"><img src="images/genmexlogo.jpg" alt="Gen Mex Concert Logo"></a>
+        <nav class="nav-menu">
+        <ul class="nav-options">
+            <li class="nav-link"><a class="link-menu" href="./about.html">About</a></li>
+            <li class="nav-link"><a class="link-menu" href="#">Program</a></li>
+            <li class="nav-link"><a class="link-menu" href="#">Join</a></li>
+            <li class="nav-link"><a class="link-menu" href="#">Sponsor</a></li>
+            <li class="nav-link"><a class="link-menu" href="#">News</a></li>
+            <li class="nav-link"><a id="gmcampaign" class="link-menu" href="#">GM Campaign</a></li>
+        </ul>
+        </nav>
+        </div>
+    </div>
+    <header class="principal-info">
+        <div class="mobile-bar display-none">
+            <nav class="mobile-navmenu">
+            <ul class="mobile-nav-options">
+                <li class="mobile-nav-link"><a class="link-menu" href="./about.html">About</a></li>
+                <li class="mobile-nav-link"><a class="link-menu" href="#">Program</a></li>
+                <li class="mobile-nav-link"><a class="link-menu" href="#">Join</a></li>
+                <li class="mobile-nav-link"><a class="link-menu" href="#">Sponsor</a></li>
+                <li class="mobile-nav-link"><a class="link-menu" href="#">News</a></li>
+                <li class="mobile-nav-link"><a class="link-menu" href="#">GM Campaign</a></li>
+                <li class="mobile-nav-link"><i class="fa-solid fa-xmark"></i></li>
+            </ul>
+            </nav>
```
- [x] You have exerted brilliance and shown great skills to ship this page :wink: but really, I feel that there are ways you could make this even more professional. As a principle  "Don't Repeat Yourself" - DRY, I suggest you reduce your lines of code by using the same `nav` items for both views. This will show your skills to achieve more with less code.
- [x] Also, I think to improve accessibility on your page, since there is no logo ono the mobile screen (to redirect back to homepage), you can add `Home` to your mobile navigation or just add the logo to the top-right corner.
- [x] Below is a design of your mobile menu :point_down: 
![image](https://user-images.githubusercontent.com/99715640/219253958-29370054-ed92-4f78-b80c-08833fcda9dd.png)

- The nav items seem too cluttered and lack adequate space. This might be difficult for user to read or click appropriately. To improve this user experience, I think you should pull out your `gap`, `padding` , or `margin`  tool to make sure that they are clearly readable and easy to interact with (click). Users with fat fingers will love this more :wink: :smile:

edgardsoza replied:
solved

AbayomiOlaoye commented on index.html:
```diff
@@ -0,0 +1,144 @@
+        </div>
+        <div class="main-content">
+        <i id="hamburger" class="fa-solid fa-bars fa-2x"></i>
+        <h3 class="hello-concert">"Hello! Welcome Gen Mex"</h1>
+        <h1 class="concert-title"><b>GEN MEX CONCERT<br>AMAZON ORIGINALS<br> 2023</b></h1>
+        <p class="concert-paragraph"> A joyful event at the NeueHouse Hollywood.
+        The event brings together a new generation of mexican fresh stars who will delight 
+        the audience with some of their best hits.
+        </p>
+        <p class="date-and-time">2023.03.03(FRI) ~ 05(SUN)</p>
+        <p class="place"><b>@NeueHouse Hollywood, Columbia Square, USA</b></p>
```
- Awesomely done job!
- [ ] Let's make this even better by doing the following after considering this portrait :point_down: 

|Your Design| Figma Design|
|:--------------:|:-----------------:|
|![image](https://user-images.githubusercontent.com/99715640/219254599-3c5f8612-9820-42e4-9dcd-72102b241de8.png)|![image](https://user-images.githubusercontent.com/99715640/219254712-fe127160-ff77-43bc-b03b-6b8681f38573.png)|

- [ ] From the above, for more accessibility advantages and user experience, I think you should increase the `line-height` of your paragraph for readability.
- [ ] If you can, kindly use the same font for the main title. The name is `cocogoose` not `lato`. I trust you will like to make this look even more appealing and identical to the benchmark.
- [ ] In my opinion, this part doesn't really form parts of your page header. By this, I meant it should not be nested in the header but in your `main` section. This is good for SEO merits.

edgardsoza replied:
done

AbayomiOlaoye commented on index.html:
```diff
@@ -0,0 +1,144 @@
+        <div class="main-container">
+            <div class="main-container-icon">
+                <i class="fa-solid fa-microphone fa-5x"></i>               
+            </div>
+            <h3 class="title-main-container">Opening</h3>
+            <p class="description-main-container">The youngest singers from Mexico will start this night with their awesome talent</p>
+        </div>
+        <div class="main-container">
+            <div class="main-container-icon">
+                <i class="fa-solid fa-user fa-5x"></i>            
+            </div>
+            <h3 class="title-main-container">Main Artists</h3>
+            <p class="description-main-container">Three out of our main artists for today will make us enjoy for 2 hours of their most popular songs</p>
+        </div>
+        <div class="main-container">
+            <div class="main-container-icon">
+                <i class="fa-solid fa-bacon fa-5x"></i>
+            </div>
+            <h3 class="title-main-container">Break</h3>
+            <p class="description-main-container">In this time you will enjoy with your friends while listening to diverse music, courtesy of the DJ. Let's enjoy</p>
+        </div>
+        <div class="main-container">
+            <div class="main-container-icon">
+                <i class="fa-solid fa-microphone fa-5x"></i>               
+            </div>
+            <h3 class="title-main-container">Opening</h3>
+            <p class="description-main-container">Another group of the youngest singers from Mexico will re-start this night with their awesome talent</p>
+        </div>
+        <div class="main-container">
+            <div class="main-container-icon">
+                <i class="fa-solid fa-user fa-5x"></i>            
+            </div>
+            <h3 class="title-main-container">Closing</h3>
+            <p class="description-main-container">The last main artists for today will make us enjoy for 2 hours of their most popular songs. Let's close this night!!</p>
+        </div>
+        </div>
```
- [x] This section looks absolutely sparkling on the desktop view but in the mobile range, things seem too cluttered with no uniform spacing or padding within the `div`, and everything seems to be floating without restraint. Check the image below and please find out how to improve this:

![image](https://user-images.githubusercontent.com/99715640/219256378-0597202c-2bd3-4581-b49c-f01b8c74d427.png)

edgardsoza replied:
solved

AbayomiOlaoye commented on index.html:
```diff
@@ -0,0 +1,144 @@
+            <p class="description-main-container">The last main artists for today will make us enjoy for 2 hours of their most popular songs. Let's close this night!!</p>
+        </div>
+        </div>
+        <a class="see-more display-none" href="#">SEE THE WHOLE PROGRAM</a>
```
- [x] This link is more like a button and a link and it increases accessibility if you can allow it to refer users to the `about.html` when clicked. 
- [x] MOreso, do you think you can use a `cursor: pointer` on this? IIt improves user experience on your page.

edgardsoza replied:
Done

AbayomiOlaoye commented on index.html:
```diff
@@ -0,0 +1,144 @@
+            </ul>
+            </nav>
+        </div>
+        <div class="main-content">
+        <i id="hamburger" class="fa-solid fa-bars fa-2x"></i>
+        <h3 class="hello-concert">"Hello! Welcome Gen Mex"</h1>
+        <h1 class="concert-title"><b>GEN MEX CONCERT<br>AMAZON ORIGINALS<br> 2023</b></h1>
+        <p class="concert-paragraph"> A joyful event at the NeueHouse Hollywood.
+        The event brings together a new generation of mexican fresh stars who will delight 
+        the audience with some of their best hits.
+        </p>
+        <p class="date-and-time">2023.03.03(FRI) ~ 05(SUN)</p>
+        <p class="place"><b>@NeueHouse Hollywood, Columbia Square, USA</b></p>
+        </div>
+    </header>
```
- [x] Have you thought about how nicely it will improve your aesthetics if you include a great `background-image`? Kindly add one please, I think your users will like it.

edgardsoza replied:
done

AbayomiOlaoye commented on styles.css:
```diff
@@ -0,0 +1,776 @@
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    margin-top: 3rem;
+    height: 20rem;
+    width: 100%;
+    margin-bottom: 3rem;
+  }
+
+  .footer-container {
+    display: flex;
+    flex-direction: row;
+    justify-content: center;
+    align-items: center;
+    text-align: center;
+    width: 80%;
+  }
+
+  .footer-logo-about {
+    width: 10rem;
+    height: 8rem;
+  }
+
+  .license-about {
+    height: 5rem;
+    font-size: 1.5rem;
+    width: 32rem;
+    color: #fff;
+    padding-top: 1.2rem;
+  }
+
+  .speakers-container {
+    display: grid;
+    grid-template-columns: repeat(2, 50%);
+    justify-content: center;
+    gap: 2rem;
+    width: 90%;
+    margin-bottom: 7rem;
+  }
+}
```
- Your is generally clean, and I love this! :love_you_gesture: but I'd like to suggest that you create a class for each `css property` that is used more than ones so you only declare it once and then allow your class to act as a reference. I see different `flex` properties and `font` properties being repeated more than twice. This will make your code DRY and cleaner with shorter code.

edgardsoza replied:
will solve in the future

AbayomiOlaoye reviewed (changes requested):
Hi @edgardsoza :+1: , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

![so_close](https://user-images.githubusercontent.com/99715640/211950260-0e3881c0-9d4f-4ea2-a670-9ef45a23a0f5.gif)

## Highlights
- Descriptive Pull Request message :heavy_check_mark: 
- Good use of GitHub flow :heavy_check_mark: 
- Meaningful commit messages :heavy_check_mark: 
- No linter error :heavy_check_mark: 
- Nice design :heavy_check_mark: 



## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

edgardsoza commented:
This is the next revision for my capston project, please note that the other reviewer requested the changes as suggested, I could
skip them if I want, but I tried to apply as many as I can, but project is functional and the changes are just optional and design 
oriented. 

"You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._"
1.  I wish I was able to get a screenshot of the strange white space after this footer section in the desktop view but strangely, it's not working. Kindly check to improve your design by removing this with space 👍
Resolved. It was a margin-bottom. 
2.  You have exerted brilliance and shown great skills to ship this page 😉 but really, I feel that there are ways you could make this even more professional. As a principle "Don't Repeat Yourself" - DRY, I suggest you reduce your lines of code by using the same nav items for both views. This will show your skills to achieve more with less code.
This is used because the two navs behave different based on the version and the hamburger button. I will try it, but not for now. 
3.  Also, I think to improve accessibility on your page, since there is no logo ono the mobile screen (to redirect back to homepage), you can add Home to your mobile navigation or just add the logo to the top-right corner.
I added this one, but as a link when user is in the about section.
4. The nav items seem too cluttered and lack adequate space. This might be difficult for user to read or click appropriately. To improve this user experience, I think you should pull out your gap, padding , or margin tool to make sure that they are clearly readable and easy to interact with (click). Users with fat fingers will love this more 😉 😄
Solved, I applied gap and adjust width to content in the options. 
5.  From the above, for more accessibility advantages and user experience, I think you should increase the line-height of your paragraph for readability.
Done.
6. If you can, kindly use the same font for the main title. The name is cocogoose not lato. I trust you will like to make this look even more appealing and identical to the benchmark.
Done. 
7. In my opinion, this part doesn't really form parts of your page header. By this, I meant it should not be nested in the header but in your main section. This is good for SEO merits.
I will not change this one for now, for me main sections starts in main program, so I will fix it in the future. 
8.  This section looks absolutely sparkling on the desktop view but in the mobile range, things seem too cluttered with no uniform spacing or padding within the div, and everything seems to be floating without restraint. Check the image below and please find out how to improve this:
Done, forgot to style it first time. 
9.  This link is more like a button and a link and it increases accessibility if you can allow it to refer users to the about.html when clicked.
Done.
10. MOreso, do you think you can use a cursor: pointer on this? IIt improves user experience on your page.
Done
11.  Have you thought about how nicely it will improve your aesthetics if you include a great background-image? Kindly add one please, I think your users will like it.
Done. Added image with details, there was one already. 
12. Your is generally clean, and I love this! 🤟 but I'd like to suggest that you create a class for each css property that is used more than ones so you only declare it once and then allow your class to act as a reference. I see different flex properties and font properties being repeated more than twice. This will make your code DRY and cleaner with shorter code.
I will also do this one in the future.

IngDavidTM commented on index.html:
```diff
@@ -0,0 +1,144 @@
+    <main class="main-program">
+        <h3 class="main-program-title">Main Program</h3>
+        <hr class="breakline">
+        <div class="main-container-cards">
+        <div class="main-container">
+            <div class="main-container-icon">
+                <i class="fa-solid fa-microphone fa-5x"></i>               
+            </div>
+            <h3 class="title-main-container">Opening</h3>
+            <p class="description-main-container">The youngest singers from Mexico will start this night with their awesome talent</p>
+        </div>
+        <div class="main-container">
+            <div class="main-container-icon">
+                <i class="fa-solid fa-user fa-5x"></i>            
+            </div>
+            <h3 class="title-main-container">Main Artists</h3>
+            <p class="description-main-container">Three out of our main artists for today will make us enjoy for 2 hours of their most popular songs</p>
+        </div>
+        <div class="main-container">
+            <div class="main-container-icon">
+                <i class="fa-solid fa-bacon fa-5x"></i>
+            </div>
+            <h3 class="title-main-container">Break</h3>
+            <p class="description-main-container">In this time you will enjoy with your friends while listening to diverse music, courtesy of the DJ. Let's enjoy</p>
+        </div>
+        <div class="main-container">
+            <div class="main-container-icon">
+                <i class="fa-solid fa-microphone fa-5x"></i>               
+            </div>
+            <h3 class="title-main-container">Opening</h3>
+            <p class="description-main-container">Another group of the youngest singers from Mexico will re-start this night with their awesome talent</p>
+        </div>
+        <div class="main-container">
+            <div class="main-container-icon">
+                <i class="fa-solid fa-user fa-5x"></i>            
+            </div>
+            <h3 class="title-main-container">Closing</h3>
+            <p class="description-main-container">The last main artists for today will make us enjoy for 2 hours of their most popular songs. Let's close this night!!</p>
+        </div>
+        </div>
```
- [ ] While you did a good job staying each card inside your 'Main program' section :+1:. As mentioned before you should align all the elements to the middle of the card to match the Figma template as you can see below.

- [OPTIONAL] Kindly check the background of each one of the cards to match with the Figma :+1: 

**Your design**

![Screenshot from 2023-02-16 08-38-41](https://user-images.githubusercontent.com/105117832/219380989-533d8c91-f4f5-4907-a124-8b3c14c372d4.png)

**Figma**

![Screenshot from 2023-02-16 08-44-10](https://user-images.githubusercontent.com/105117832/219381220-72c4bde1-a2c5-426a-a744-e6c9ae656b79.png)

edgardsoza replied:
done

IngDavidTM commented on index.html:
```diff
@@ -0,0 +1,144 @@
+        </nav>
+        </div>
+    </div>
+    <header class="principal-info">
+        <div class="mobile-bar display-none">
+            <nav class="mobile-navmenu">
+            <ul class="mobile-nav-options">
+                <li class="mobile-nav-link"><a class="link-menu" href="./about.html">About</a></li>
+                <li class="mobile-nav-link"><a class="link-menu" href="#">Program</a></li>
+                <li class="mobile-nav-link"><a class="link-menu" href="#">Join</a></li>
+                <li class="mobile-nav-link"><a class="link-menu" href="#">Sponsor</a></li>
+                <li class="mobile-nav-link"><a class="link-menu" href="#">News</a></li>
+                <li class="mobile-nav-link"><a class="link-menu" href="#">GM Campaign</a></li>
+                <li class="mobile-nav-link"><i class="fa-solid fa-xmark"></i></li>
+            </ul>
+            </nav>
```
- [OPTIONAL] I think it would be better if the `x` button is set in the top right of your nav bar menu :+1: because it would improve the User experience.
- [OPTIONAL] Also, kindly make sure that your nav bar covers the screen and the user can not access other sections when it is open :+1: 

**Your design when you open the nave bar**

![Screenshot from 2023-02-16 08-48-23](https://user-images.githubusercontent.com/105117832/219383062-ae1af7bc-596d-4c74-b824-de8e0daebc68.png)

edgardsoza replied:
Done X button, mobile menu view 100%.

IngDavidTM reviewed (changes requested):
## Hi @edgardsoza  🙋‍♂️, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

<img src="https://media2.giphy.com/media/31Q36EHJ7C5ZoIMw79/giphy.gif"/>

## Top highlight 🎯
- The Pull Request follows the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

edgardsoza commented:
This is the next revision for my capston project, please note that the other reviewer requested the changes as suggested, I could
skip them if I want, but I tried to apply as many as I can, but project is functional and the changes are just optional and design 
oriented. 

"You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._"
1.  I wish I was able to get a screenshot of the strange white space after this footer section in the desktop view but strangely, it's not working. Kindly check to improve your design by removing this with space 👍
Resolved. It was a margin-bottom. 
2.  You have exerted brilliance and shown great skills to ship this page 😉 but really, I feel that there are ways you could make this even more professional. As a principle "Don't Repeat Yourself" - DRY, I suggest you reduce your lines of code by using the same nav items for both views. This will show your skills to achieve more with less code.
This is used because the two navs behave different based on the version and the hamburger button. I will try it, but not for now. 
3.  Also, I think to improve accessibility on your page, since there is no logo ono the mobile screen (to redirect back to homepage), you can add Home to your mobile navigation or just add the logo to the top-right corner.
I added this one, but as a link when user is in the about section.
4. The nav items seem too cluttered and lack adequate space. This might be difficult for user to read or click appropriately. To improve this user experience, I think you should pull out your gap, padding , or margin tool to make sure that they are clearly readable and easy to interact with (click). Users with fat fingers will love this more 😉 😄
Solved, I applied gap and adjust width to content in the options. 
5.  From the above, for more accessibility advantages and user experience, I think you should increase the line-height of your paragraph for readability.
Done.
6. If you can, kindly use the same font for the main title. The name is cocogoose not lato. I trust you will like to make this look even more appealing and identical to the benchmark.
Done. 
7. In my opinion, this part doesn't really form parts of your page header. By this, I meant it should not be nested in the header but in your main section. This is good for SEO merits.
I will not change this one for now, for me main sections starts in main program, so I will fix it in the future. 
8.  This section looks absolutely sparkling on the desktop view but in the mobile range, things seem too cluttered with no uniform spacing or padding within the div, and everything seems to be floating without restraint. Check the image below and please find out how to improve this:
Done, forgot to style it first time. 
9.  This link is more like a button and a link and it increases accessibility if you can allow it to refer users to the about.html when clicked.
Done.
10. MOreso, do you think you can use a cursor: pointer on this? IIt improves user experience on your page.
Done
11.  Have you thought about how nicely it will improve your aesthetics if you include a great background-image? Kindly add one please, I think your users will like it.
Done. Added image with details, there was one already. 
12. Your is generally clean, and I love this! 🤟 but I'd like to suggest that you create a class for each css property that is used more than ones so you only declare it once and then allow your class to act as a reference. I see different flex properties and font properties being repeated more than twice. This will make your code DRY and cleaner with shorter code.
I will also do this one in the future.
New comments:
1.  While you did a good job staying each card inside your 'Main program' section 👍. As mentioned before you should align all the elements to the middle of the card to match the Figma template as you can see below.
Done. 
2. [OPTIONAL] Kindly check the background of each one of the cards to match with the Figma 👍
There is no FIGMA to match design. This is the best I found for this. 
3. [OPTIONAL] I think it would be better if the x button is set in the top right of your nav bar menu 👍 because it would improve the User experience.
[OPTIONAL] Also, kindly make sure that your nav bar covers the screen and the user can not access other sections when it is open 👍
Done X button, mobile menu view 100%.

binhussen commented on about.html:
```diff
@@ -0,0 +1,118 @@
+        <div class="mobile-bar display-none">
+            <nav class="mobile-navmenu-about">
+            <ul class="mobile-nav-options">
+                <li class="mobile-nav-link"><i class="fa-solid fa-xmark"></i></li>
```
-[Optional] Kindly it would be better if you move the close `X` button to the top right for having better user experience👍
you can use `display: absolute` to achieve this 👇
```
.fa-solid{
    position: absolute;
    color: white;
    top: -10px;
    right: -10px;
}
```

binhussen reviewed (approved):
Hi @edgardsoza , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
<img src="https://user-images.githubusercontent.com/65852621/133696182-6c5b784b-0083-427a-b24d-3ec870915f79.jpg"/>
## Highlights
- All linter checks passed ✔️
- Descriptive pull request ✔️
- Used GitHub flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
